src/index.js

```diff
@@ -35,10 +44,13 @@ export default function (Vue, {
           if (this._persistWatchers.indexOf(name) === -1) {
             this._persistWatchers.push(name)
 
-            this.$watch(name, val => {
-              store.data[name] = val
-              write(storeName, JSON.stringify(store))
-            }, { deep: true })
+            this.$watch(name, {
+              handler: val => {
+                store.data[name] = val;
+                write(storeName, JSON.stringify(store));
+              },
+              deep: true
+            })
```

Comment:
this is incorrect, `handler` is for component option `watch` instead of the `$watch` method.